x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/StringFieldScript.java

```diff
@@ -41,6 +41,22 @@
         StringFieldScript newInstance(LeafReaderContext ctx);
     }
 
+    public static final Factory PARSE_FROM_SOURCE = (field, params, lookup) -> (LeafFactory) ctx -> new StringFieldScript(
+        field,
+        params,
+        lookup,
+        ctx
+    ) {
+        @Override
+        public void execute() {
+            for (Object v : extractFromSource(field)) {
+                if (v != null) {
+                    emit(v.toString());
```

Comment:
do we need to check for the result of toString? emit(null) will cause an error and we are not catching exceptions here

Reply:
If `toString` returns null we're in a sad place.

Reply:
true. I got carried away I think :)